ql/src/java/org/apache/hadoop/hive/ql/parse/QBSubQuery.java

```diff
@@ -606,7 +606,8 @@ void subqueryRestrictionsCheck(RowResolver parentQueryRR,
     /*
      * Restriction.14.h :: Correlated Sub Queries cannot contain Windowing clauses.
      */
-    if (  hasWindowing && hasCorrelation) {
+    if (operator.getType() != SubQueryType.EXISTS && operator.getType() != SubQueryType.NOT_EXISTS &&
+        hasWindowing && hasCorrelation) {
       throw new CalciteSubquerySemanticException(ASTErrorUtils.getMsg(
           ErrorMsg.UNSUPPORTED_SUBQUERY_EXPRESSION.getMsg(),
           subQueryAST, "Correlated Sub Queries cannot contain Windowing clauses."));
```

Comment:
Can we update this message as well as the comment below (I think it is outdated)?

Reply:
Updated the error message.
Did you mean the comment above? `Restriction.14.h :: Correlated Sub Queries cannot contain Windowing clauses.`

Reply:
I think `Restriction.13.m` is also out-of-date based on your latest work (not only related to this patch)?

Reply:
Yes. Thanks for pointing out. Created follow-up patch of #2039: #2118